dhcpv6/option_bootfileparam_test.go

```diff
@@ -0,0 +1,31 @@
+package dhcpv6
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testBootfileParam = "\x00\x30initrd=http://myserver.mycompany.local/initrd.xz\x00\x0eroot=/dev/sda1\x00\x02rw\x00\x24netconsole=..:\000:.something\000here.::.."
+)
+
+func TestOptBootFileParam(t *testing.T) {
+	expected := testBootfileParam
+	opt, err := ParseOptBootFileParam([]byte(expected))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(opt.ToBytes()) != expected {
+		t.Fatalf("Invalid boot file parameter. Expected %v, got %v", expected, opt)
+	}
+}
+
+func TestParsedTypeOptBootFileParam(t *testing.T) {
+	opt, err := ParseOption(OptionBootfileParam, []byte(testBootfileParam))
+	require.NoError(t, err)
+	bootfileParamOpt, ok := opt.(OptBootFileParam)
+	require.True(t, ok, fmt.Sprintf("invalid type: %T instead of %T", opt, bootfileParamOpt))
+	require.Equal(t, testBootfileParam, string(bootfileParamOpt.ToBytes()))
```

Comment:
Testing that the round-trip is unchanged is good, but it would also be nice to test that the parsed value is correct. If for example the option were parsing as a single string (as the original implementation did) instead of 4, this test would pass when it should fail.

Reply:
Yeah, a good thought.